wireapp/wire-android
# feat: Custom server no network dialog #WPB-11627

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11627" title="WPB-11627" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-11627</a>  [Android] Show information to user when he tries to login, but no network is available
  </td></table>
  <br />
 


https://wearezeta.atlassian.net/browse/WPB-11627

# What's new in this PR?

### Issues

When we're trying to connect with custom server and we didnt have internet connection, we didnt display correct popup

### Solutions

Due to the matter of handling invalid json being exactly the same as no network, we have to check on our own if we're connected to internet or not. Depending on that we display no network dialog.

### Testing

#### How to Test

- Turn off the internet
- Launch deeplink, for example: adb shell am start -a android.intent.action.VIEW -d "wire://access/?config=https://nginz-https.bund-next-column-offline-android.wire.link/deeplink.json"
- Popup should be visible
- Turn on the internet and tap try again
- Configuration for server should be fetched

### Attachments (Optional)

![Screenshot_20241023_073149_Wire Beta](https://github.com/user-attachments/assets/14a9cac0-d061-46f9-ba2c-086ba5c795f4)

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
